modules/api-compatibility-common-terminology.adoc

```diff
@@ -0,0 +1,52 @@
+// Module included in the following assemblies:
+//
+// * rest_api/understanding-compatibility-guidelines.adoc
+
+[id="api-compatibility-common-terminology_{context}"]
+= API compatibility common terminology
+
+[id="api-compatibility-common-terminology-api_{context}"]
+== Application Programming Interface (API)
+
+An API is a public interface implemented by a software program that enables it to interact with other software. In {product-title}, the API is served from a centralized API server and is used as the hub for all system interaction.
+
+[id="api-compatibility-common-terminology-aoe_{context}"]
+== Application Operating Environment (AOE)
```

Comment:
@deads2k @derekwaynecarr isn't there a third layer to consider beyond API and AOE, namely things like node-level behavior, networking behavior, and such?  Is that considered part of the AOE (even if you use hostnetworking, or are otherwise interacting at lower levels w/ the kubelet, network, or storage layers on the node?)

or are we stating that if you're doing those things we make no guarantees?

Reply:
(AOE doesn't seem to cover it because the description of AOE below is pretty clearly constrained to the container environment)

Reply:
the AOE is intended to capture stuff that is expressed on the pod specification (sysctls, envars, secrets, etc.) but not intended to extend to things like a CNI choice that may vary per installation.  active/passive extensions to the product may influence a test-matrix for a solution independent of the guarantees provided in this document.

Reply:
i'm adding a sentence or two related to this, but most if it will be covered by the additional doc derek has coming around test matrices.